lib/TimeSeries/AdaptiveFilter.pm

```diff
@@ -0,0 +1,210 @@
+package TimeSeries::AdaptiveFilter;
+
+=head1 NAME
+
+TimeSeries::AdaptiveFilter - Adaptive filter for data stream with possible outliers
+
+=cut
+
+our $VERSION = '0.02';
+
+=head1 VERSION
+
+Version 0.02
+
+=head1 STATUS
+
+=begin HTML
+
+<p>
+    <a href="https://travis-ci.org/binary-com/perl-TimeSeries-AdaptiveFilter"><img src="https://travis-ci.org/binary-com/perl-TimeSeries-AdaptiveFilter.svg" /></a>
+</p>
+
+=end HTML
+
+=cut
+
+use strict;
+use warnings;
+
+use List::Util qw(sum max);
+
+use parent qw/Exporter/;
```

Comment:
consider this code (designed to be run with `-l`):

```perl
use TimeSeries::AdaptiveFilter 'filter';
print "filter is defined? ".(defined &filter ? "yes" : "no");
print "ivan is defined? ".(defined &ivan ? "yes" : "no");

print "can $_: ".TimeSeries::AdaptiveFilter->can($_)
    for (qw/import as_heavy export export_fail export_to_level export_tags export_ok_tags require_version fritz/);
```

You'd expect the first line to say `yes` the next `no`, right? The `filter` function gets imported but there is no such thing as an `ivan` function.

Also, you'd probably expect that `TimeSeries::AdaptiveFilter` can do `import`. That's the function needed to handle our `@EXPORT_OK`, right?

How about the rest? I guess `TimeSeries::AdaptiveFilter->can('fritz')` is supposed to return false, right?

But would you expect that your module has a method called `as_heavy`? Yet, it has. It's inherited from `Exporter`.

If you change to `use Exporter 'import'`, you get exactly what you want, no surprises. You get a module with a working `import` function, so that it adheres to all those `@EXPORT` variables. This method is also documented. What is not documented are the surprising consequences of the first method.

To be frank, I won't merge anything that inherits from `Exporter` unless the author explains in detail why he needs exactly that.